cmd/abapEnvironmentCreateTag.go

```diff
@@ -0,0 +1,246 @@
+	requestBodyJson, err := json.Marshal(&requestBodyStruct)
+	if err != nil {
+		return err
+	}
+
+	log.Entry().Debugf("Request body: %s", requestBodyJson)
+	resp, err := abaputils.GetHTTPResponse("POST", con, requestBodyJson, client)
+	if err != nil {
+		err = abaputils.HandleHTTPError(resp, err, errorMessage, con)
+		return err
+	}
+	defer resp.Body.Close()
+
+	// Parse response
+	var body CreateTagResponse
+	var abapResp map[string]*json.RawMessage
+	bodyText, _ := ioutil.ReadAll(resp.Body)
+
+	json.Unmarshal(bodyText, &abapResp)
+	json.Unmarshal(*abapResp["d"], &body)
+
+	con.URL = con.Host + "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull(guid'" + body.UUID + "')"
+	err = checkStatus(con, client, com)
+
+	if err == nil {
+		log.Entry().Info("Created tag " + requestBodyStruct.Tag + " for repository " + requestBodyStruct.RepositoryName + " with commitID " + requestBodyStruct.CommitID)
+	} else {
+		log.Entry().Error("NOT created: Tag " + requestBodyStruct.Tag + " for repository " + requestBodyStruct.RepositoryName + " with commitID " + requestBodyStruct.CommitID)
+	}
+
+	return err
+}
+
+func checkStatus(con abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, com abaputils.Communication) (err error) {
+	status := "R"
+	pollIntervall := com.GetPollIntervall()
+	count := 0
+	for {
+		count += 1
+		entity, _, err := abaputils.GetStatus("Could not create Tag", con, client)
+		if err != nil {
+			return err
+		}
+		status = entity.Status
+		if status != "R" {
+			if status == "E" {
+				err = errors.New("Could not create Tag")
+			}
+			return err
+		}
+		if count >= 200 {
+			return errors.New("Could not create Tag (Timeout)")
+		}
+		time.Sleep(pollIntervall)
+	}
+}
+
+func prepareBacklog(config *abapEnvironmentCreateTagOptions) (backlog []CreateTagBacklog, err error) {
+
+	descriptor, err := abaputils.ReadAddonDescriptor(config.Repositories)
+	if err != nil {
+		return
+	}
+	repos := descriptor.Repositories
+
+	for _, repo := range repos {
+
+		backlogInstance := CreateTagBacklog{repositoryName: repo.Name, commitID: repo.CommitID}
+		if repo.VersionYAML != "" {
+			tag := Tag{tagName: "v" + repo.VersionYAML, tagDescription: "Generated by the ABAP Environment Pipeline"}
+			backlogInstance.tags = append(backlogInstance.tags, tag)
+		}
+		backlog = append(backlog, backlogInstance)
+	}
+
+	if config.RepositoryName != "" && config.CommitID != "" {
+		backlog = append(backlog, CreateTagBacklog{repositoryName: config.RepositoryName, commitID: config.CommitID})
+	}
+
+	if config.CreateTagForAddonProductVersion && descriptor.AddonProduct != "" && descriptor.AddonVersionYAML != "" {
+		addonProductDash := strings.Replace(descriptor.AddonProduct, "/", "-", 2)
+		backlog = addTagToList(backlog, addonProductDash+"-"+descriptor.AddonVersionYAML, "Generated by the ABAP Environment Pipeline")
```

Comment:
see my coment in the documentation section .... I guess this will also write the addon product version for the repo which was manually specified and not part of the addon descriptor config file...